app/components/telegram-auth.tsx

```diff
@@ -271,10 +280,9 @@ export default function TelegramAuth() {
       if (result instanceof Api.auth.AuthorizationSignUpRequired) {
         throw new Error('user needs to sign up')
       }
-      if (process.env.NODE_ENV === 'production') {
-        if (BigInt(result.user.id.toString()) !== BigInt(user?.id ?? 0)) {
-          throw new Error('login user and user using the app must match')
-        }
+      // Verify the authenticated user matches the Mini App user
```

Comment:
did we test this works?

Reply:
I did yeah - saw it kick me back out with a reasonable error message when I logged in with the wrong account with or without 2fa